scripts/new_python_based_image.py

```diff
@@ -0,0 +1,555 @@
+
+
+def process_paths(copied_paths, source_version, target_version):
+    """
+    Processes the list of copied paths by replacing Python versions and running pipenv lock on Pipfiles.
+
+    Args:
+        copied_paths (list): The list of copied paths to process.
+        source_version (str): The source Python version.
+        target_version (str): The target Python version.
+    """
+    if copied_paths.count == 0:
+        LOGGER.info("No paths to process.")
+        return
+
+    for path in copied_paths:
+        if not os.path.exists(path):
+            LOGGER.warning(f"The path '{path}' does not exist.")
+            continue
+
+        replace_version_in_directory(path, source_version, target_version)
+        process_pipfiles(path, target_version)
+
+
+def process_pipfiles(path, target_version):
+    """
+    Processes Pipfiles in a given path by running `pipenv lock` on them.
+
+    Args:
+        path (str): The path to search for Pipfiles.
+        target_version (str): The target Python version to use with `pipenv lock`.
+    """
+    for root, _, files in os.walk(path):
+        for file_name in files:
+            if file_name.startswith("Pipfile") and "lock" not in file_name:
+                pipfile_path = os.path.join(root, file_name)
+                run_pipenv_lock(pipfile_path, target_version)
+
+
+def run_pipenv_lock(pipfile_path, target_version):
+    """
+    Runs `pipenv lock` for a specified Pipfile to generate a new lock file.
+
+    Args:
+        pipfile_path (str): The path to the Pipfile.
+        target_version (str): The target Python version to use with `pipenv lock`.
+    """
+    LOGGER.info(f"Running pipenv lock for '{pipfile_path}'")
+    env = os.environ.copy()
+    env["PIPENV_PIPFILE"] = os.path.basename(pipfile_path)
+
+    try:
+        result = subprocess.run(
+            ["pipenv", "lock", "--python", target_version],
+            cwd=os.path.dirname(pipfile_path),
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            env=env
+        )
+        LOGGER.debug(result.stdout.decode())
+    except subprocess.CalledProcessError as e:
+        LOGGER.error(f"Error running pipenv lock for '{pipfile_path}'")
+        LOGGER.debug(e.stderr.decode())
+
+
+def manual_checks():
+    """
+    Provides a list of manual checks to perform after the script execution.
+
+    Returns:
+        list: A list of manual checks to review.
+    """
+    return [
+        "Check the issues thrown during the script execution, if any, and fix them manually.",
+        "Check if the Python version replacements have been performed correctly on the new files.",
+        "Review and make the appropriate changes in the Makefile and CI-related files.",
+        "Push the changes to a new branch on your fork to build the new images with GitHub workflows.",
+        "Test the new images."
+    ]
+
+
+def main():
+    """
+    Main function to execute the script.
+    """
```

Comment:
```suggestion
    """
    Main function to execute the script.
    """
```

this would be overcommenting, for sure